Лабораторная работа 1

GhosT-FlexAgen opened the pull request.

GhosT-FlexAgen committed:
Add files via upload

GhosT-FlexAgen committed:
Add files via upload

dmitrylaabe commented on Лабораторная работа 1/task_1.py:
```diff
@@ -0,0 +1,8 @@
+missed_number_id = 4
+
+# Замена значений пропущенного элемента средним арифметическим
+numbers_uniq = numbers[:missed_number_id]+numbers[missed_number_id+1:]
```
Можно обойтись без вспомогательного списка

dmitrylaabe commented on Лабораторная работа 1/task_3.py:
```diff
@@ -0,0 +1,11 @@
+lines = 50
+chars = 25
+
+book_size = pages * lines * chars * 4
```
Используйте константы для всех числовых значений!

GhosT-FlexAgen committed:
Исправление ошибок с учетом комментариев

Task 1 
- Устранен вспомогательный список numbers_uniq

Task 2
- Созданы константы для всех числовых значений

dmitrylaabe commented on Лабораторная работа 1/task_1.py:
```diff
@@ -2,7 +2,9 @@
 # Замена значений пропущенного элемента средним арифметическим
-numbers_uniq = numbers[:missed_number_id]+numbers[missed_number_id+1:]
-numbers[missed_number_id] = sum(numbers_uniq) / (len(numbers_uniq) + 1)
+summ_numbers = sum(numbers[:missed_number_id]+numbers[missed_number_id+1:])
```
Здесь эффективнее использовать sum() + sum(), иначе создается вспомогательный список перед суммированием.

dmitrylaabe commented on Лабораторная работа 1/task_1.py:
```diff
@@ -2,7 +2,9 @@
-numbers_uniq = numbers[:missed_number_id]+numbers[missed_number_id+1:]
-numbers[missed_number_id] = sum(numbers_uniq) / (len(numbers_uniq) + 1)
+summ_numbers = sum(numbers[:missed_number_id]+numbers[missed_number_id+1:])
+count_numbers = (len(numbers[:missed_number_id]+numbers[missed_number_id+1:]) + 1)
```
верно, но проще разделить на количество элементов в исходном списке: len(numbers)

GhosT-FlexAgen committed:
Улучшение решения задания 1 в соответствии с комментариями

- sum() заменена на sum() + sum(), чтобы не создавался вспомогательный список перед суммированием
- подсчёт количества элементов в списке заменен на len(numbers)
